### PR TITLE
Video capture

### PR DESCRIPTION
**Pull request by YushraJewon**
Allow plugin to capture video.

### CODE REVIEW

File: src/android/CameraPreviewFragment.java [hunk 1]
```diff
@@ -24,8 +28,18 @@
 import androidx.camera.core.ImageCaptureException;
 import androidx.camera.core.Preview;
 import androidx.camera.lifecycle.ProcessCameraProvider;
+import androidx.camera.video.FileOutputOptions;
+import androidx.camera.video.MediaStoreOutputOptions;
```
Comment (zfir) [inline src/android/CameraPreviewFragment.java:32]:
```suggestion
```

File: src/android/SimpleCameraPreview.java [hunk 2]
```diff
@@ -246,6 +249,49 @@ public void fetchLocation() {
         }
     }
 
+    private boolean captureVideo(boolean useFlash, CallbackContext callbackContext) {
+        if (fragment == null) {
+            callbackContext.error("Camera is closed");
+            return true;
+        }
+
+        fragment.captureVideo( useFlash, new VideoCallback() {
```
Comment (zfir) [inline src/android/SimpleCameraPreview.java:258]:
```suggestion
        fragment.captureVideo(useFlash, new VideoCallback() {
```

File: src/android/CameraPreviewFragment.java [hunk 3]
```diff
@@ -232,6 +264,61 @@ public void hasFlash(HasFlashCallback hasFlashCallback) {
         hasFlashCallback.onResult(camera.getCameraInfo().hasFlashUnit());
     }
 
+    public void captureVideo(boolean useFlash, VideoCallback videoCallback) {
+       if (torchActivated) {
+           useFlash = true;
+       } else {
+           camera.getCameraControl().enableTorch(useFlash);
+       }
```
Comment (zfir) [inline src/android/CameraPreviewFragment.java:272]:
```suggestion
         if (torchActivated) {
             useFlash = true;
         } else {
             camera.getCameraControl().enableTorch(useFlash);
         }
```

Some lines not indented properly xD

File: src/android/CameraPreviewFragment.java [hunk 4]
```diff
@@ -124,31 +146,41 @@ public void startCamera() {
...
-            camera = cameraProvider.bindToLifecycle(
-                    this,
-                    cameraSelector,
-                    preview,
-                    imageCapture
-            );
-        } catch (IllegalArgumentException e) {
-            // Error with result in capturing image with default resolution
-            e.printStackTrace();
+        this.getActivity().runOnUiThread(() -> {
```
Comment (zfir) [inline src/android/CameraPreviewFragment.java:159]:
```suggestion
        this.getActivity().runOnUiThread(() -> {
```

Why do we need to run this on UI thread?

File: src/android/CameraPreviewFragment.java [hunk 5]
```diff
@@ -232,6 +264,61 @@ public void hasFlash(HasFlashCallback hasFlashCallback) {
...
+
+// Configure the recorder to save the video to the specified file
+        FileOutputOptions outputOptions = new FileOutputOptions.Builder(videoFile).build();
+
+        recording = videoCapture.getOutput()
+                .prepareRecording(this.getContext().getApplicationContext(), outputOptions)
+                .withAudioEnabled()
+                .start(ContextCompat.getMainExecutor(this.getContext()), videoRecordEvent -> {
+                    if (videoRecordEvent instanceof VideoRecordEvent.Start) {
+                        videoCallback.onStart(null,true, null);
```
Comment (zfir) [inline src/android/CameraPreviewFragment.java:299]:
```suggestion
                        videoCallback.onStart(null, true, null);
```

File: src/android/CameraPreviewFragment.java [hunk 6]
```diff
@@ -102,9 +124,9 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         return containerView;
     }
 
+
```
Comment (parveshneedhoo) [inline src/android/CameraPreviewFragment.java:127]:
```suggestion
```

File: src/android/CameraPreviewFragment.java [hunk 7]
```diff
@@ -124,31 +146,41 @@ public void startCamera() {
...
                     this,
                     cameraSelector,
                     preview,
-                    imageCapture
+                    imageCapture,
+                    videoCapture
             );
-        }
+            }
+
```
Comment (parveshneedhoo) [inline src/android/CameraPreviewFragment.java:182]:
```suggestion
```

**Review by parveshneedhoo — COMMENTED** [review]
For audio permission deny part, we can treat it same as we do for image picker.

Before starting video capture :
1)we first request permission
2)then we check if it has permission
3) if it has, we start recording withAudioEnabled else start it without

**Pull request closed without merging**